rpc/debug_api.go

```diff
@@ -0,0 +1,109 @@
+package rpc
+
+import (
+	"context"
+	"io/ioutil"
+	"time"
+	"os"
+	"net/http"
+	"net/url"
+	"encoding/json"
+	"errors"
+
+	"github.com/sirupsen/logrus"
+	"github.com/patrickmn/go-cache"
+
+	web3Types "github.com/openweb3/web3go/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/scroll-tech/rpc-gateway/util/rpc/handlers"
+)
+
+var (
+	whiteListURL     string
+	whiteListCache = cache.New(10 * time.Minute, 10 * time.Minute)
+)
+
+func init() {
+	whiteListURL = os.Getenv("WHITELIST_BACKEND_URL")
+	logrus.Info("whiteListURL: ", whiteListURL)
+}
+
+// debugAPI provides ethereum debug API.
+type debugAPI struct{}
+
+func (api *debugAPI) TraceTransaction(ctx context.Context, blockHash common.Hash, opts interface{}) (*web3Types.TraceTransactionResult, error) {
+	remoteAddr := remoteAddrFromContext(ctx)
+	logrus.Info("remoteAddr: ", remoteAddr)
+	valid, err := isIPValid(remoteAddr)
+	if err != nil {
+		return nil, err
+	}
+	if !valid {
+		logrus.Debug("Invalid IP ", remoteAddr)
+		return nil, errors.New("Operation not permitted")
+	}
+	w3c := GetEthClientFromContext(ctx)
+	return w3c.Eth.TraceTransaction(blockHash, opts)
```

Comment:
Once we move it into a new file in web3go:

```suggestion
	return w3c.Debug.TraceTransaction(blockHash, opts)
```